src/main/java/org/threeten/extra/chrono/InternationalFixedDate.java

```diff
@@ -0,0 +1,887 @@
+     */
+    public static InternationalFixedDate from(final TemporalAccessor temporal) {
+        if (temporal instanceof InternationalFixedDate) {
+            return (InternationalFixedDate) temporal;
+        }
+
+        return InternationalFixedDate.ofEpochDay(temporal.getLong(ChronoField.EPOCH_DAY));
+    }
+
+    /**
+     * Obtains a {@code InternationalFixedDate} representing a date in the International fixed calendar
+     * system from the proleptic-year and day-of-year fields.
+     * <p/>
+     * This returns a {@code InternationalFixedDate} with the specified fields.
+     * The day must be valid for the year, otherwise an exception will be thrown.
+     *
+     * @param prolepticYear the International fixed proleptic-year
+     * @param dayOfYear     the International fixed day-of-year, from 1 to 366
+     * @return the date in International fixed calendar system, not null
+     * @throws DateTimeException if the value of any field is out of range, or if the day-of-year is invalid for the year
+     */
+    static InternationalFixedDate ofYearDay(final int prolepticYear, final int dayOfYear) {
+        InternationalFixedChronology.YEAR_RANGE.checkValidValue(prolepticYear, ChronoField.YEAR_OF_ERA);
+        InternationalFixedChronology.DAY_OF_YEAR_RANGE.checkValidValue(dayOfYear, ChronoField.DAY_OF_YEAR);
+
+        boolean isLeapYear = InternationalFixedChronology.INSTANCE.isLeapYear(prolepticYear);
+        boolean isYearDay = dayOfYear == InternationalFixedChronology.DAYS_IN_YEAR + (isLeapYear ? 1 : 0);
+        boolean isLeapDay = isLeapYear && dayOfYear == 169;
+
+        if (isYearDay || isLeapDay) {
+            return of(prolepticYear, isLeapDay, isYearDay);
+        }
+
+        int doy = (isLeapYear && dayOfYear > 169) ? dayOfYear - 2 : dayOfYear - 1;
+        int dayOfMonth = 1 + (doy % InternationalFixedChronology.DAYS_IN_MONTH);
+        int month = 1 + (doy / InternationalFixedChronology.DAYS_IN_MONTH);
+
+        return of(prolepticYear, month, dayOfMonth);
+    }
+
+    /**
+     * Obtains a {@code InternationalFixedDate} representing a date in the International fixed calendar
+     * system from the epoch-day.
+     *
+     * @param epochDay the epoch day to convert based on 1970-01-01 (ISO)
+     * @return the date in International fixed calendar system, not null
+     * @throws DateTimeException if the epoch-day is out of range
+     */
+    static InternationalFixedDate ofEpochDay(final long epochDay) {
+        InternationalFixedChronology.EPOCH_DAY_RANGE.checkValidValue(epochDay, ChronoField.EPOCH_DAY);
+        long zeroDay = epochDay + InternationalFixedChronology.DAYS_0000_TO_1970;
+
+        if (zeroDay < 0) {
+            throw new DateTimeException("Invalid epoch: " + epochDay);
+        }
+
+        long year = (400 * zeroDay) / DAYS_PER_CYCLE;
+        long doy = zeroDay - (InternationalFixedChronology.DAYS_IN_YEAR * year + getLeapYearsBefore(year));
```

Comment:
Your year is an estimate, but you don't true up afterwards.  I'm not sure how safe the multiplication by 400 first is - you do have tests around boundaries for non-leap centuries, right?